python/lsst/summit/testing/analysis/m1m3/inertia_compensation_system.py

```diff
@@ -452,55 +438,29 @@ def evaluate_m1m3_ics_single_slew(
     InertiaCompensationSystemAnalysis
         Object containing the results of the analysis.
     """
-    logger.info("Retrieving TMA slew event.")  # type: ignore
+    log = log.getChild(__name__) if log is not None else logging.getLogger(__name__)
+
+    log.info("Retrieving TMA slew event.")  # type: ignore
     event = get_tma_slew_event(day_obs, seq_number)
 
-    logger.info("Start inertia compensation system analysis.")  # type: ignore
+    log.info("Start inertia compensation system analysis.")  # type: ignore
     event = get_tma_slew_event(day_obs, seq_number)
-    logger.info("Start inertia compensation system analysis.")  # type: ignore
+    log.info("Start inertia compensation system analysis.")  # type: ignore
     performance_analysis = M1M3ICSAnalysis(
         event,
         inner_pad=inner_pad,
         outer_pad=outer_pad,
         n_sigma=n_sigma,
-        logger=logger,
+        log=log,
     )
 
     return performance_analysis
 
 
-def create_logger(name: str) -> logging.Logger:
-    """
-    Create a logger object with the specified name and returns it.
-
-    Parameters
-    ----------
-    name : str
-        The name of the logger object.
-
-    Returns
-    -------
-    logger : logging.Logger
-        The logger object with the specified name.
-    """
-    formatter = logging.Formatter(
-        "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
-    )
-    formatter.datefmt = "%Y-%m-%d %H:%M:%S"
-
-    handler = logging.StreamHandler()
-    handler.setLevel(logging.ERROR)
-    handler.setFormatter(formatter)
-
-    logger = logging.getLogger(name)
-    logger.setLevel(logging.DEBUG)
-
-    return logger
-
-
 if __name__ == "__main__":
-    logger = create_logger("M1M3ICSAnalysis")
-    logger.info("Start")
-    results = evaluate_m1m3_ics_single_slew(20230802, 38, logger=logger)
+    log = create_logger("M1M3ICSAnalysis")
+    log.info("Start")
+    event_maker = TMAEventMaker()
```

Comment:
Moving that here does fix my problem, but I don't think it fixes yours... I _think_ this will work when run from the command line as a script, but people importing those functions still won't have them work unless they're in a notebook with object already declared, and with that exact name. I am not sure why you don't just make it an arg to the function.

Reply:
And I'm still worried about why that's not giving you a linting error, too.

Reply:
I applied a few changes to address what you mentioned above. I could have passed the whole event maker into the class, but I preferred to pass only what was strictly necessary.